service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/kubernetesNamespace/CreateKubernetesNamespaceStep.java

```diff
@@ -40,7 +40,10 @@ public StepResult doStep(FlightContext context) throws InterruptedException, Ret
         context
             .getWorkingMap()
             .get(ControlledResourceKeys.AZURE_CLOUD_CONTEXT, AzureCloudContext.class);
-    var clusterResource = kubernetesClientProvider.getClusterResource(workspaceId);
+    var clusterResource =
+        kubernetesClientProvider
+            .getClusterResource(workspaceId)
+            .orElseThrow(() -> new RuntimeException("No shared cluster found"));
```

Comment:
This is the same exception that was being thrown in the call, before - just moved to the call site instead.